tests/python/test_pointwise.py

```diff
@@ -421,3 +421,40 @@ def fusion_func(fd: FusionDefinition):
 
     with pytest.raises(RuntimeError, match="No executor supports provided fusion."):
         _ = fd.execute(inputs)
+
+
+def test_pointwise_issue_3512():
+    inputs = [
+        torch.testing.make_tensor(
+            (1, 2048, 512), dtype=torch.bfloat16, device="cuda:0"
+        ),
+    ]
+
+    # T34 and T198 are both candidate for reference tv in pointwise scheduler.
+    # We can only pick T198 for scheduling though, because a expanded dimension
+    # is merged by the reshape that produces T198, which means transformation
+    # on T34 wouldn't be able to propagate from T192 to T198.
+    def fusion_func(fd: FusionDefinition):
+        T3 = fd.define_tensor(
+            shape=[1, 2048, 512],
+            contiguity=[None, True, True],
+            dtype=DataType.BFloat16,
+            is_cpu=False,
+            stride_order=[2, 1, 0],
+        )
+        T33 = fd.ops.reshape(T3, new_shape=[1, 2048, 8, 64])
+        T34 = fd.ops.permute(T33, dims=[0, 2, 1, 3])
+        T185 = fd.ops.broadcast_in_dim(
+            T34, shape=[1, 8, 1, 2048, 64], broadcast_dims=[0, 1, 3, 4]
+        )
+        T192 = fd.ops.broadcast_in_dim(
+            T185, shape=[1, 8, 4, 2048, 64], broadcast_dims=[0, 1, 2, 3, 4]
+        )
+        T198 = fd.ops.reshape(T192, new_shape=[1, 32, 2048, 64])
+        fd.add_output(T34)
+        fd.add_output(T198)
+
+    with FusionDefinition() as fd:
+        fusion_func(fd)
+
+    _ = fd.execute(inputs)
```

Comment:
How are we confirming the test is indeed scheduled as we intend?

Reply:
I don't think we need to check that, since it's currently running as two segmented kernel. Do you think we need to check for a less then 2 segments thing? I can convert this to a cpp test and check that.

The test is for functional check, since ToT throws an assert on this example. I thought a functional check is good enough.

Reply:
I'm not sure what exactly is considered "functional". It doesn't seem to even validate the actual results, but what we can say from this test is that there's no explicit error like `NVF_ERROR`.

Reply:
That's a good call. I should have added a reference implementation on this one.

I think I should instead move it to a cpp test where I can use FusionDefinition for validation. :bow: